### PR TITLE
[PLGN-629] Active Directory LDAP - Fix problem where some ASCII characters were not escaped properly

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Fix problem where some ASCII characters were not escaped properly
  - Refactored code

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/active_directory_ldap/help.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Example input:
 
 #### Add User
   
-Adds the specified Active Directory user
+This action is used to adds the specified Active Directory user
```
Comment (ablakley-r7) [inline plugins/active_directory_ldap/help.md:60]:
```suggestion
This action is used to add the specified Active Directory user
```

File: plugins/active_directory_ldap/help.md [hunk 2]
```diff
@@ -107,7 +107,7 @@ Example output:
 
 #### Delete
   
-Deletes the LDAP object specified
+This action is used to deletes the LDAP object specified
```
Comment (ablakley-r7) [inline plugins/active_directory_ldap/help.md:110]:
```suggestion
This action is used to delete the LDAP object specified
```

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
Looks good, couple of changes with the new description format.

File: plugins/active_directory_ldap/komand_active_directory_ldap/util/api.py [hunk 3]
```diff
@@ -176,15 +176,21 @@ def manage_users(self, conn, dns: List[str], status: bool) -> dict:
...
                 ADUtils.change_account_status(conn, dn, status, self.logger)
                 successes.append(dn)
-            except Exception as exception:
-                failures.append({"dn": dn, "error": str(exception)})
-                self.logger.error(f"Error: Failed to modify user {dn}")
+            except Exception:
+                try:
+                    escaped_dn = ADUtils.escape_non_ascii_characters(dn)
+                    ADUtils.change_account_status(conn, escaped_dn, status, self.logger)
+                    successes.append(escaped_dn)
```
Comment (joneill-r7) [inline plugins/active_directory_ldap/komand_active_directory_ldap/util/api.py:190]: would it be better to append `dn` so it reads as expected to the user?

File: plugins/active_directory_ldap/komand_active_directory_ldap/actions/enable_user/action.py [hunk 4]
```diff
@@ -14,4 +15,12 @@ def __init__(self):
...
     def run(self, params={}):
-        return {Output.SUCCESS: self.connection.client.manage_user(params.get(Input.DISTINGUISHED_NAME), True)}
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        distinguished_name = params.get(Input.DISTINGUISHED_NAME)
+        # END INPUT BINDING - DO NOT REMOVE
+
+        try:
+            return {Output.SUCCESS: self.connection.client.manage_user(distinguished_name, True)}
+        except PluginException:
+            self.logger.info("Escaping non-ascii characters...")
```
Comment (joneill-r7) [inline plugins/active_directory_ldap/komand_active_directory_ldap/actions/enable_user/action.py:25]: is escape not called within manage_user meaning we should need this try/except?

**Review by ablakley-r7 — APPROVED** [review]

**Review by joneill-r7 — APPROVED** [review]

**Pull request merged**